### PR TITLE
Fix pivot subtotals loading

### PR DESCRIPTION
**Pull request by b0bi79**
ClosedXML does not correctly define the properties of subtotals for records, like this: 
```
<pivotField showAll="0">
...
<dataField name="Sum Amount paid" fld="6" subtotal="sum" baseField="0" baseItem="0"/>
```
In this case, Excel shows subtotals, but after loading into ClosedXML and then saving the document, the subtotals are turned off.

### CODE REVIEW

**Comment by b0bi79** [comment]
This PR is needed for the release of ClosedXML.Report

**Comment by igitur** [comment]
Do you have a unit test for this?

**Comment by igitur** [comment]
I also notice that if I apply these commits on top of #1196 's branch, then it breaks #1196 's unit tests. The generated pivot table in `TwoPivotWithOneSourceTest` changes from not having subtotals to having subtotals. Not sure whether this intended.

**Comment by b0bi79** [comment]
Today I can't, but tomorrow I will check and add new tests.

**Comment by igitur** [comment]
Please rebase this on `develop`. With #1196 now merged, I think that you will now already have a test case for this.

**Comment by igitur** [comment]
You should have rebased, not merged.

**Comment by b0bi79** [comment]
I tried to rebase, but apparently did something wrong. How can I cancel this?

**Comment by igitur** [comment]
I'll do it and push to your branch a bit later.  I don't want you to accidentally delete the work you've done.

**Comment by b0bi79** [comment]
Sorry, I don't work well with git. Thank you.

**Comment by igitur** [comment]
image

This is how I rebased on `develop`.

**Comment by igitur** [comment]
Do a `git pull origin/fix_pivot_subtotals_loading --rebase` on your side.

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 1]
```diff
@@ -749,7 +749,9 @@ private void LoadSpreadsheetDocument(SpreadsheetDocument dSpreadsheet)
 
                                         if (df.BaseField?.Value != null)
                                         {
-                                            var col = pt.SourceRange.Column(df.BaseField.Value + 1);
+                                            IXLRangeColumn col = df.BaseField.Value == 0
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Load.cs]: Can you please explain this?

Comment (b0bi79) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: I saw this issue directly in this test. The source file has
		`<x:dataField name="Sum Amount paid" fld="6" subtotal="sum" showDataAs="normal" baseField="3" baseItem="0" numFmtId="0" />`
but saved file
		`<x:dataField name="Sum Amount paid" fld="6" subtotal="sum" showDataAs="normal" baseField="4" baseItem="0" numFmtId="0" />`

Comment (igitur) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: So if `df.BaseField.Value == 0`, then you would be referencing `pt.SourceRange.Column(1)` (the first column), but if `df.BaseField.Value == 1`, then you would *also* be reference  `pt.SourceRange.Column(1)`?

**Comment by igitur** [comment]
Please unhide all sheets in the reference files. Hidden sheets add to confusion ;-)

**Comment by igitur** [comment]
Also try to simplify `Sheet1` in order to keep the file as small as possible. For example, we don't need the image in the first row.

File: ClosedXML/Excel/XLWorkbook_Load.cs (hunk 1 above)
Comment (igitur) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: I had to go read up on what `BaseField` means. As per https://docs.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.datafield?view=openxml-2.8.1 it's used when the "`ShowDataAs` calculation is in use". So in your reference file, this attribute isn't relevant. But in the case where it is relevant, this change is wrong and should be reverted.

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 2]
```diff
@@ -2852,7 +2851,7 @@ private static void GenerateTableDefinitionPartContent(TableDefinitionPart table
                     var baseField = pt.SourceRange.Columns().FirstOrDefault(c => c.Cell(1).Value.ToInvariantString() == value.BaseField);
                     if (baseField != null)
                     {
-                        df.BaseField = baseField.ColumnNumber() - 1;
+                        df.BaseField = baseField.ColumnNumber() - pt.SourceRange.RangeAddress.FirstAddress.ColumnNumber;
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Save.cs]: I pushed this commit to fix your base field issue. Please `git pull --rebase` the changes on your side.

File: ClosedXML/Excel/XLWorkbook_Load.cs (hunk 1 above)
Comment (b0bi79) [reply ClosedXML/Excel/XLWorkbook_Load.cs]: But on the other hand, it is also not true that opening and saving a file changes this attribute. It should not be.

File: ClosedXML/Excel/XLWorkbook_Save.cs (hunk 2 above)
Comment (b0bi79) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: Thanks

**Comment by igitur** [comment]
I rewrote the commit history and pushed to your branch. If the CI tests pass, I will merge.

**Comment by igitur** [comment]
Thanks for the contribution. We're making good progress now.

**Pull request merged**